packages/core/src/lib/message/version_0.ts

```diff
@@ -119,12 +120,19 @@ export class Encoder implements IEncoder {
  * messages.
  */
 export function createEncoder({
-  pubsubTopic = DefaultPubsubTopic,
+  pubsubTopicShardInfo,
   contentTopic,
   ephemeral,
   metaSetter
 }: EncoderOptions): Encoder {
-  return new Encoder(contentTopic, ephemeral, pubsubTopic, metaSetter);
+  return new Encoder(
+    contentTopic,
+    ephemeral,
+    pubsubTopicShardInfo?.index
```

Comment:
why would the index not be defined? It's not optional on `SingleTopicShardInfo`.

Reply:
actually updated this part of #1742